### ISSUE TITLE
[SIM906] Merge nested os.path.join calls

### ISSUE THREAD

**Issue by Skylion007** [MartinThoma/flake8-simplify#101]
## Explanation

Explain briefly why you think this makes the code simpler.
* This is a common but fun one I have noticed. I have noticed this pattern occur a lot and I believe in most cases it can be simplified to a single function call. I guess this would be kinda of similar to the isinstance checks where people don't realize the function can consume multiple args.

## Example

```python
a = "/a"
b = "b"
c = "c"
# Bad
os.path.join(a,os.path.join(b,c))

# Good
os.path.join(a,b,c)
```

**Comment by MartinThoma** [comment]
I didn't know that this was possible! I'll add this when I have more time for sure - this should also be easy / have little potential for false-positives. Very nice suggestion!

**Comment by MartinThoma** [comment]
Side-note: [pathlib.Path](https://docs.python.org/3/library/pathlib.html) should be used more often. If a, b, and c were Path objects, you could simply do:

```
d = a / b / c
```

**Comment by MartinThoma** [comment]
Many usages of os.path can be replaced by `pathlib`.

**Comment by Skylion007** [comment]
@MartinThoma True, but there can be performance issues with using Pathlib. For instance, I think the black formatter  [removed / reworked](https://github.com/psf/black/issues/1950) a lot of their Pathlib usage back to os.path in their library due to performance issues.

**Comment by MartinThoma** [comment]
```
$ astpretty --no-show-offsets /dev/stdin <<< `cat example.txt`
Module(
    body=[
        Expr(
            value=Call(
                func=Attribute(
                    value=Attribute(
                        value=Name(id='os', ctx=Load()),
                        attr='path',
                        ctx=Load(),
                    ),
                    attr='join',
                    ctx=Load(),
                ),
                args=[
                    Name(id='a', ctx=Load()),
                    Call(
                        func=Attribute(
                            value=Attribute(
                                value=Name(id='os', ctx=Load()),
                                attr='path',
                                ctx=Load(),
                            ),
                            attr='join',
                            ctx=Load(),
                        ),
                        args=[
                            Name(id='b', ctx=Load()),
                            Name(id='c', ctx=Load()),
                        ],
                        keywords=[],
                    ),
                ],
                keywords=[],
            ),
        ),
    ],
    type_ignores=[],
)
```
